src/wasm-lib/src/wasm.rs

```diff
@@ -23,54 +15,32 @@ pub async fn execute_wasm(
     memory_str: &str,
     id_generator_str: &str,
     units: &str,
-    engine_manager: kcl_lib::engine::conn_wasm::EngineCommandManager,
-    fs_manager: kcl_lib::fs::wasm::FileSystemManager,
+    engine_manager: kcl_lib::wasm_engine::EngineCommandManager,
+    fs_manager: kcl_lib::wasm_engine::FileSystemManager,
     project_directory: Option<String>,
     is_mock: bool,
 ) -> Result<JsValue, String> {
     console_error_panic_hook::set_once();
-    // deserialize the ast from a stringified json
 
-    let program: Node<Program> = serde_json::from_str(program_str).map_err(|e| e.to_string())?;
-    let memory: kcl_lib::executor::ProgramMemory = serde_json::from_str(memory_str).map_err(|e| e.to_string())?;
-    let id_generator: kcl_lib::executor::IdGenerator =
-        serde_json::from_str(id_generator_str).map_err(|e| e.to_string())?;
-    let units = kcl_lib::settings::types::UnitLength::from_str(units).map_err(|e| e.to_string())?;
-
-    let engine: std::sync::Arc<Box<dyn kcl_lib::engine::EngineManager>> = if is_mock {
-        Arc::new(Box::new(
-            kcl_lib::engine::conn_mock::EngineConnection::new()
-                .await
-                .map_err(|e| format!("{:?}", e))?,
-        ))
-    } else {
-        Arc::new(Box::new(
-            kcl_lib::engine::conn_wasm::EngineConnection::new(engine_manager)
-                .await
-                .map_err(|e| format!("{:?}", e))?,
-        ))
-    };
-    let fs = Arc::new(kcl_lib::fs::FileManager::new(fs_manager));
-    let context_type = if is_mock {
-        kcl_lib::executor::ContextType::Mock
+    let program = Program::from_json(program_str)?;
```

Comment:
I'm pretty sure this changed. Before it deserialized into `Node<Program>`. Now it deserializes into a struct with a single field `ast: Node<Program>`. I don't see any TS changes that would make this change on the other side of the WASM boundary.

Reply:
Yeah, I thought it would need a lot more on the TS side, I'm both surprised and a bit afraid